appinventor/schemekit/src/YailMatrix.swift

```diff
@@ -0,0 +1,310 @@
+//
+//  YailMatrix.swift
+//  SchemeKit
+//
+//  Created by Grant Hu on 4/1/25.
+//  Copyright © 2025 Massachusetts Institute of Technology. All rights reserved.
+//
+@objc open class YailMatrix: NSObject, SCMObject {
```

Comment:
We may want to add a property to the SCMValue protocol `isMatrix`, which should be true for this class and false for all others.